src/Google/Ads/GoogleAds/V0/Util/ErrorUtils.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\V0\Util;
+
+use \Google\Protobuf\Any;
+use \Google\Protobuf\Internal\Message;
+use \Google\Rpc\Status;
+use Google\Ads\GoogleAds\V0\Errors\GoogleAdsFailure;
+use Google\Ads\GoogleAds\V0\Errors\ErrorLocation\FieldPathElement;
+
+
+/** 
+ * Contains utility methods for handling partial failure of operations. 
+ */
+ final class ErrorUtils
+ {
+    /**
+     * Gets a list of all partial failure error messages for a given response. Operations are 
+     * indexed from 0.
+     *
+     * <p>For example, given the following GoogleAdsFailure:
+     *
+     * <pre>
+     *   <code>
+     *     errors {
+     *       message: "Too low."
+     *       location {
+     *         field_path_elements {
+     *           field_name: "operations"
+     *           index {
+     *             value: 1
+     *           }
+     *         }
+     *         field_path_elements {
+     *           field_name: "create"
+     *         }
+     *         field_path_elements {
+     *           field_name: "campaign"
+     *         }
+     *       }
+     *     }
+     *     errors {
+     *       message: "Too low."
+     *       location {
+     *         field_path_elements {
+     *           field_name: "operations"
+     *           index {
+     *             value: 2
+     *           }
+     *         }
+     *         field_path_elements {
+     *           field_name: "create"
+     *         }
+     *         field_path_elements {
+     *           field_name: "campaign"
+     *         }
+     *       }
+     *     }
+     *   </code>
+     * </pre>
+     *
+     * A single GoogleAdsError instance would be returned for operation index 1 and 2, and an empty
+     * list otherwise.
+     *
+     * @param int operationIndex the index of the operation, starting from 0.
+     * @param \Google\Rpc\Status partialFailureStatus a partialFailure status, with the detail list 
+     *  containing GoogleAdsFailure instances.
+     * @return Google\Ads\GoogleAds\V1\Errors\GoogleAdsError[] an array containing the 
+     *  GoogleAdsError instances for a given operation index.
+     */
+    public static function getGoogleAdsErrorsFromStatus(
+        int $operationIndex,
```

Comment:
Is this a valid PHP 7.1 syntax?

Reply:
Yep scalar type hinting has been introduced in PHP 7: http://php.net/manual/en/migration70.new-features.php